### PR TITLE
fluid-synth 2.0.3

### PR DESCRIPTION
**Pull request by ColinKinloch**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
5 dependent failures, which probably require a revision bump:

```
brew-test-bot.sierra.linkage --test audacious
brew-test-bot.sierra.linkage --test carla
brew-test-bot.sierra.test carla
brew-test-bot.sierra.linkage --test dosbox-x
brew-test-bot.sierra.test dosbox-x
brew-test-bot.sierra.linkage --test mpd
brew-test-bot.sierra.test mpd
brew-test-bot.sierra.linkage --test scummvm
brew-test-bot.sierra.test scummv
```

**Comment by ColinKinloch** [comment]
It's a major version with API changes, should this be a new formula; `fluid-synth2.rb`?

File: Formula/fluid-synth.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ class FluidSynth < Formula
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
   depends_on "glib"
+  depends_on "dbus" => :optional
```
Comment (fxcoudert) [inline Formula/fluid-synth.rb]: No new option please. What would be the user benefit of this new dependency?

File: Formula/fluid-synth.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ class FluidSynth < Formula
...
   depends_on "libsndfile" => :optional
```
Comment (fxcoudert) [inline Formula/fluid-synth.rb]: Lett's make these two dependencies mandatory.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
No new formula, just bump the revision number of the formulas I mentioned (see the `revision` items in some formulas), so that they get rebuilt with the new version of `fluid-synth`.

File: Formula/fluid-synth.rb (hunk 1 above)
Comment (ColinKinloch) [reply Formula/fluid-synth.rb]: It's to request realtime status from rtkit, but since rtkit seems to only support linux, you're correct. I'll remove it.

**Comment by ColinKinloch** [comment]
`carla` throws the error
```
OSError: dlopen(/usr/local/Cellar/carla/1.9.11/lib/carla/libcarla_standalone2.dylib, 6): Library not loaded: /usr/local/opt/fluid-synth/lib/libfluidsynth.1.dylib
  Referenced from: /usr/local/Cellar/carla/1.9.11/lib/carla/libcarla_standalone2.dylib
  Reason: image not found
```
This is because the dylib is installed to `/usr/local/opt/fluid-synth/lib/libfluidsynth.2.dylib`

**Comment by fxcoudert** [comment]
> bump the revision number of the formulas I mentioned (see the `revision` line in some formulas)

This. See for example how it's done in https://github.com/Homebrew/homebrew-core/pull/33011

**Comment by fxcoudert** [comment]
So now, `carla dosbox-x mpd` don't build anymore 😢 

- `dosbox-x`:

```
In file included from midi.cpp:127:
./midi_synth.h:175:3: error: no matching function for call to 'fluid_set_log_function'
                fluid_set_log_function(FLUID_PANIC, synth_log, NULL);
                ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/fluidsynth/log.h:76:22: note: candidate function not viable: no known conversion from 'void (int, char *, void *)' to 'fluid_log_function_t' (aka 'void (*)(int, const char *, void *)') for 2nd argument
fluid_log_function_t fluid_set_log_function(int level, fluid_log_function_t fun, void *data);
                     ^
```

- `mpd` has the same:

```
src/decoder/plugins/FluidsynthDecoderPlugin.cxx:83:2: error: no matching function for call to 'fluid_set_log_function'
        fluid_set_log_function(LAST_LOG_LEVEL,
        ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/fluid-synth/2.0.1/include/fluidsynth/log.h:76:22: note: candidate function not viable: no known conversion from 'void (int, char *, void *)' to 'fluid_log_function_t' (aka 'void (*)(int, const char *, void *)') for 2nd argument
fluid_log_function_t fluid_set_log_function(int level, fluid_log_function_t fun, void *data);
                     ^
```

- `carla`:

```
CarlaPluginFluidSynth.cpp:80:40: error: use of undeclared identifier 'FLUID_REVERB_DEFAULT_ROOMSIZE'
        fluid_synth_set_reverb(fSynth, FLUID_REVERB_DEFAULT_ROOMSIZE, FLUID_REVERB_DEFAULT_DAMP, FLUID_REVERB_DEFAULT_WIDTH, FLUID_REVERB_DEFAULT_LEVEL);
                                       ^
```

**Comment by SMillerDev** [comment]
@ColinKinloch do you know if the tools mentioned by @fxcoudert have new versions for this version of fluid-synth yet?

**Comment by ColinKinloch** [comment]
* carla 2.0 has fixes for this, is in [release candidate](http://kxstudio.linuxaudio.org/News/?action=view&url=carla-20-rc2-is-here) so should be due soon.
* mpd is fixed as of v0.21.3 #34182
* dosbox-x needs `char *` needs to be switched out with `const char *` joncampbell123/dosbox-x#935

**Comment by igas** [comment]
Go away bot, we are tracking blockers here.

**Comment by ColinKinloch** [comment]
The blockers are resolved. This is ready to be pulled.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Once everything else is updated, it turns out to be a simple update in the end :) I don't see a reason for holding this.

@ColinKinloch and everyone here: thanks for your work on this!

**Comment by fxcoudert** [comment]
One detail @ColinKinloch: we normally ask for one commit per formula (i.e., one commit for the main update and one per revision bump). I'm not going to ask you to rebase here, given the amount of time already spent in this PR, but I thought I'd mention it anyway :)

Thanks again for your work

**Pull request closed without merging**